### PR TITLE
*: supports batch split

### PR DESCRIPTION
**Pull request by nolouch**

### CODE REVIEW

File: server/cluster_worker.go [hunk 1]
```diff
@@ -77,6 +71,55 @@ func (c *RaftCluster) handleAskSplit(request *pdpb.AskSplitRequest) (*pdpb.AskSp
 	return split, nil
 }
 
+func (c *RaftCluster) validRequestRegion(reqRegion *metapb.Region) error {
+	startKey := reqRegion.GetStartKey()
+	region, _ := c.GetRegionByKey(startKey)
```
Comment (disksing) [inline server/cluster_worker.go]: Can `region` be nil?

File: server/cluster_worker.go [hunk 2]
```diff
@@ -77,6 +71,55 @@ func (c *RaftCluster) handleAskSplit(request *pdpb.AskSplitRequest) (*pdpb.AskSp
...
+
+		peerIDs := make([]uint64, len(request.Region.Peers))
+		for i := 0; i < len(peerIDs); i++ {
+			if peerIDs[i], err = c.s.idAlloc.Alloc(); err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+
+		// Disable merge for the 2 regions in a period of time.
+		c.coordinator.mergeChecker.RecordRegionSplit(reqRegion.GetId())
```
Comment (disksing) [inline server/cluster_worker.go]: So is `reqRegion` recorded multiple times?

**Comment by disksing** [comment]
LGTM.

**Comment by nolouch** [comment]
PTAL @disksing @Connor1996 @rleungx

File: server/cluster_worker.go [hunk 3]
```diff
@@ -93,6 +139,24 @@ func (c *RaftCluster) checkSplitRegion(left *metapb.Region, right *metapb.Region
...
+		return errors.New("invalid split region")
+	}
+
+	for i := 1; i < len(regions); i++ {
+		left := regions[i-1]
+		right := regions[i]
+		if !bytes.Equal(left.GetEndKey(), right.GetStartKey()) {
+			return errors.New("invalid split region")
+		}
+		if len(right.GetEndKey()) != 0 && bytes.Compare(left.GetStartKey(), right.GetEndKey()) > 0 {
```
Comment (rleungx) [inline server/cluster_worker.go]: Do you mean `>=` here?

**Comment by disksing** [comment]
PTAL @rleungx

**Comment by Connor1996** [comment]
Please update kvproto

**Pull request merged**